kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigData.java

```diff
@@ -38,7 +38,7 @@ public class ProxyConfigData {
 
     private static String toYaml(Object filterDefs) {
         try {
-            return CONFIG_OBJECT_MAPPER.writeValueAsString(filterDefs);
+            return CONFIG_OBJECT_MAPPER.writeValueAsString(filterDefs).stripTrailing();
```

Comment:
should add a comment about the trailing newline making it easier for us to break the test expectations